runtime/lua/vim/diagnostic.lua

```diff
@@ -32,6 +32,9 @@ local M = {}
 --- The source of the diagnostic
 --- @field source? string
 ---
+--- The kind of the diagnostic's message
+--- @field message_kind lsp.MarkupKind
```

Comment:
@gpanders are you good with referring to lsp types within the vim.Diagnostic structure?

Reply:
I think that's fine. The "types" are just interfaces, in particular they are "structural type", not nominal types. So sharing them every which way seems perfectly fine to me.

Reply:
We've thus far avoided it successfully. Typically what we try to do is shove any LSP-specific stuff into the `lsp/diagnostic.lua` module and keep `vim.diagnostic` LSP-free. If it's just a type definition though it's probably not a big deal, and it probably doesn't make sense to replicate this type.

An alternative could be to make this a simple `boolean` (e.g. `@field plaintext boolean` or `@field markdown boolean`). That of course wouldn't extend if other markup types are added in the future though.

Reply:
We could loosen the relationship slightly with something like:

```lua
--- @alias vim diagnostic.MarkupKind LSP.MarkupKind
```

Then in the future we _could_ make the `vim.diagnostic` version a superset.